### DISCUSSION TITLE
What does "Closing dead socket" msg mean?

### DISCUSSION THREAD

**Discussion by xjtupdy** [fortio/fortio#895] [Q&A] [answered]
When I make http traffic load from fortioclient to fortioserver, I make Istio changes that will drain HTTP Listeners. Then I get msgs from fortioclient as belows. All http requests get 200 response code finally. 

`
{"ts":1709685862.875530,"level":"info","r":100,"file":"http_client.go","line":1070,"msg":"Closing dead socket (err at first read)","err":"read tcp 10.53.158.166:49936->172.28.237.47:8080: read: connection reset by peer","thread":5,"run":0}
`

If I use grpc traffic, there are not any msgs when I make the same Istio changes.

I want to know the reason of these msgs and whether it  will affect my http request.

**Comment by ldemailly** [comment, marked as answer]
So it's juts a warning/info telling your that the other side closed the socket (despite keep alive) - which is a legal thing to do for any server but still interesting - for instance in kubernetes this tends to happen when one of the pods is terminated (for instance when the istio side car gracefully drains connection during shutdown due to rollout or pod changes)

**Reply by xjtupdy** [reply]
Thanks for your reply. 
Why there is no such info when using grpc traffic load？

**Reply by ldemailly** [reply]
re no such thing in grpc, I am not sure entirely actually, I suspect it's because grpc internally doesn't bother mentioning socket closes as they are technically not errors / are expected
